pkg/controller/certificates/trigger/trigger_controller.go

```diff
@@ -193,6 +190,23 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 	return nil
 }
 
+// shouldBackoffReissuingOnFailure tells us if we should back off from
+// reissuing the certificate and for how much time.
+func shouldBackoffReissuingOnFailure(log logr.Logger, c clock.Clock, crt *cmapi.Certificate) (backoff bool, delay time.Duration) {
+	if crt.Status.LastFailureTime == nil {
+		return false, 0
+	}
+
+	now := c.Now()
+	durationSinceFailure := now.Sub(crt.Status.LastFailureTime.Time)
+	if durationSinceFailure > retryAfterLastFailure {
```

Comment:
if these are both 0 we still give a true but a 0 time. I think we should make this `>=`

Reply:
Ooh good catch!! I added a test case for that too